### PR TITLE
Confusion matrix branch

### PR DESCRIPTION
**Pull request by swkasula**
Related issues
De-serializing the ConfMatrices: Seq[Seq[Long]] resulting in ClassCastException as there was no @JsonDeserialize
annotation used for this attribute: https://github.com/salesforce/TransmogrifAI/pull/533/files

Describe the proposed solution
Fix is it update the ConfMatrices format from Seq[Seq[Long]] -> Seq[ConfusionMatrixPerThreshold] to de-serialize in the right format

Copy of: https://github.com/salesforce/TransmogrifAI/pull/545

### CODE REVIEW

File: core/src/main/scala/com/salesforce/op/evaluators/OpMultiClassificationEvaluator.scala [hunk 1]
```diff
@@ -596,6 +599,13 @@ case class MisClassificationsPerCategory
   MisClassifications: Seq[ClassCount]
 )
 
+case class ConfusionMatrixPerThreshold
+(
+  Threshold: Double,
+  @JsonDeserialize(contentAs = classOf[java.lang.Long])
+  ConfusionMatrixCounts: Seq[Long]
```
Comment (crupley) [inline core/src/main/scala/com/salesforce/op/evaluators/OpMultiClassificationEvaluator.scala:606]: lowerCamelCase on these 2 params

**Comment by feifjiang** [comment]
lgtm @swkasula

**Comment by crupley** [comment]
@swkasula can we add a test to confirm the serialization/deserialization is working? That would probably belong in the ModelSelectorSummaryTest

**Comment by swkasula** [comment]
> probably

that's a good point, will add it.

**Comment by swkasula** [comment]
> @swkasula can we add a test to confirm the serialization/deserialization is working? That would probably belong in the ModelSelectorSummaryTest

That's a good point, will add it.

File: core/src/main/scala/com/salesforce/op/evaluators/OpMultiClassificationEvaluator.scala (hunk 1 above)
Comment (swkasula) [reply core/src/main/scala/com/salesforce/op/evaluators/OpMultiClassificationEvaluator.scala:606]: Seems like that there was a suggestion in the past to use the upper camel case to follow conventions in that file, so leaving as-is.

**Review by crupley — APPROVED** [review]
LGTM

**Comment by swkasula** [comment]
@crupley @tovbinm @nicodv Could you please assist how these checks can be completed: 
security/snyk (Salesforce OSS )
license/snyk (Salesforce OSS

**Review by nicodv — APPROVED** [review]
LGTM

**Comment by amateurhuman** [comment]
@swkasula I removed the Snyk checks from the repo which cleared them from this PR

**Pull request merged**